stor/dx.py

```diff
@@ -387,29 +612,34 @@ def exists(self):
             return True
         except (NotFoundError, ValueError):
             pass
-        # otherwise we could be a directory, so try to grab first
-        # file/subfolder
-        if self._is_folder():
-            try:
-                self.list(limit=1)
-                return True
-            except NotFoundError:
-                return False
-        return False
+        # otherwise we could be a directory, so try to listdir folder
+        # note list doesn't error on non-existent folder and cannot be used here
+        try:
+            self.listdir()
+            return True
+        except NotFoundError:
+            return False
+
+    def getsize(self):
+        if not self.resource:
+            return self.stat()['dataUsage']*1e9
+        else:
+            return self.stat()['size']
 
     def stat(self):
-        if not self.__stat:
+        """Performs a stat on the path.
 
-            if self.canonical_resource:
-                self.__stat = dxpy.DXFile(dxid=self.canonical_resource,
-                                          project=self.canonical_project).describe()
-            else:
-                self.__stat = dxpy.DXProject(dxid=self.canonical_project).describe()
-        return self.__stat
+        Raises:
+            NotFoundError: When the project or resource cannot be found.
```

Comment:
how would `NotFoundError` get raised here? Is this implicit in the `canonical_resource` or `canonical_path` lookups? Is there an easier way to make that check explicit? It's a little confusing that this happens on a property access right now.

Reply:
Yes this is implicit in 'canonical_resource', but the error message is generic to all functions. For example a stat() on folder with trailing slash would give `ValueError : Cannot call on folder ()` and without trailing slash would give `NotFoundError : No data object was found at given path`

Reply:
I can change the docstrings to reflect this more intuitively. I can make the check explicit with try except blocks, but there's a ton of places where this is done, and it's cleaner (IMO) to have one try catch in `canonical_resource` than all over the place